### PR TITLE
Use the d8-rs-router.php for drupal 7

### PR DESCRIPTION
**Pull request by MartijnBraam**
The d8-rs-router.php from the runserver module seems to work fine for drupal-7 and fixes the issue of paths containing a dot resolving to a 404 page. See issue #1641

### CODE REVIEW

**Comment by kenorb** [comment]
I think the [test failed](https://travis-ci.org/drush-ops/drush/jobs/83312101) for unknown reason and it's not related to this PR, can the test be refreshed?

**Comment by weitzman** [comment]
Re-ran and it failed again. Agree that failure looks unrelated. Hmmmm.

**Comment by greg-1-anderson** [comment]
I've seen some folks asking about this for Drupal 7 again. Needs to be rebased against master.

**Comment by darthsteven** [comment]
@MartijnBraam any chance I can have commit access to the forked repo? I'd love to get this up to date so it could be merged in.

**Comment by MartijnBraam** [comment]
@darthsteven I have given you commit access

**Comment by darthsteven** [comment]
@MartijnBraam amazing! Thank you.

**Comment by darthsteven** [comment]
I've updated the branch to be in-sync with master, all tests are passing!

I guess it's worth noting that this will stop the overrides to `$conf` happening, because there's some magic in the Drupal 7 prepend file that aren't in the Drupal 8 version.

**Comment by darthsteven** [comment]
Right, I've re-instated the code from runserver-prepend that was doing the conf and watchdog magic.
I've also re-factored the files a little so that things are broken down into 6/7/8+ includes.

File: commands/runserver/rs-router_7.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// a convenient place to capture logs.
+if (!function_exists('system_watchdog')) {
+  // Check function_exists as a safety net in case it is added in future.
+  function system_watchdog($log_entry = array()) {
+    // Drupal <= 7.x defines VERSION. Drupal 8 defines Drupal::VERSION instead.
+    if (defined('VERSION')) {
+      $uid = $log_entry['user']->uid;
+    }
+    else {
+      $uid = $log_entry['user']->id();
```
Comment (weitzman) [inline commands/runserver/rs-router_7.php]: This looks like D8 code in a D7 file.

File: commands/runserver/runserver.drush.inc [hunk 2]
```diff
@@ -131,11 +131,15 @@ function drush_core_runserver($uri = NULL) {
...
-    $extra = ' "' . __DIR__ . '/d8-rs-router.php"';
+  if (drush_drupal_major_version() >= 8) {
+    $extra = ' "' . __DIR__ . '/rs-router.php"';
+  }
+  elseif (drush_drupal_major_version() == 7) {
+    $extra = ' "' . __DIR__ . '/rs-router_7.php"';
   }
   else {
-    $extra = ' --define auto_prepend_file="' . __DIR__ . '/runserver-prepend.php"';
+    // For Drupal 6 we prepend an include to what PHP was going invoke.
```
Comment (weitzman) [inline commands/runserver/runserver.drush.inc]: The master branch of Drush no longer supports Drupal 6. I dont mind removing this bit at merge time if that makes things simple.

**Comment by weitzman** [comment]
With this patch D8 worked fine but D7 did not. I got pages without stylesheets and the homepage was empty.

**Comment by darthsteven** [comment]
If we don't need to support Drupal 6 then let's drop that and that'll make the code much simpler.

I'll look into why it didn't work for Drupal 7 and get fixes in as required.

**Comment by larruda** [comment]
Some colleagues and I are so much interested in fixing this. @darthsteven anything I could help you with?

**Comment by jljohnstone** [comment]
I've been working on a D7 site and have been experiencing the problem with image styles not being generated. Disabling clean URLs resolves this problem but but a proper fix would be great. I'm happy to help if I can.

**Comment by acrollet** [comment]
I've added a follow-on PR in #2501

**Comment by acrollet** [comment]
update: the follow-on PR was merged, this issue can probably be closed?

**Pull request closed without merging**